tests/homework2/task3/test_smart_args.py

```diff
@@ -0,0 +1,84 @@
+import datetime
+from time import sleep
+from unittest import TestCase
+
+from homework.homework2.task3.smart_args import smart_args, Evaluated, Isolated
+
+
+class SmartArgsTestCase(TestCase):
+    def test_isolated_inside_evaluated(self):
+        with self.assertRaises(ValueError) as context:
+
+            @smart_args
+            def check(*, arg=Evaluated(Isolated())):
+                pass
+
+        self.assertTrue("Isolated cannot be used with Evaluated" in str(context.exception))
```

Comment:
Could you please add some function that can extract an exception message from context by exception type?